keras/backend/theano_backend.py

```diff
@@ -1554,7 +1554,8 @@ def relu(x, alpha=0., max_value=None):
 def softmax(x, axis=-1):
     if axis == -1 or axis == x.ndim - 1:
         return T.nnet.softmax(x)
-    return T.exp(x) / T.exp(x).sum(axis=axis, keepdims=True)
+    return T.exp(x - x.max()) / T.exp(x - x.max()
+                                      ).sum(axis=axis, keepdims=True)
```

Comment:
Style nit: consider breaking the line at `/` (with parens), or after opening a parens